pkg/sql/sem/builtins/builtins.go

```diff
@@ -3335,3 +3434,15 @@ func asJSON(d tree.Datum) (json.JSON, error) {
 		return nil, pgerror.NewErrorf(pgerror.CodeInternalError, "unexpected type %T for asJSON", d)
 	}
 }
+
+func asJSONObjectKey(d tree.Datum) (string, error) {
+	switch t := d.(type) {
```

Comment:
Would you mind moving this up to where the JSON builtins are defined, and also commenting that it handles the special case of DNull for the purposes of error messages (and any other reasons why it's used)?